src/main/java/io/vertx/ext/asyncsql/impl/ClientHelper.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ *  All rights reserved. This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License v1.0
+ *  and Apache License v2.0 which accompanies this distribution.
+ *
+ *  The Eclipse Public License is available at
+ *  http://www.eclipse.org/legal/epl-v10.html
+ *
+ *  The Apache License v2.0 is available at
+ *  http://www.opensource.org/licenses/apache2.0.php
+ *
+ *  You may elect to redistribute this code under either of these licenses.
+ */
+
+package io.vertx.ext.asyncsql.impl;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.shareddata.LocalMap;
+import io.vertx.ext.asyncsql.AsyncSQLClient;
+
+
+/**
+ *
+ * This class handles sharing the client instances by using a local shared map.
+ *
+ * Note - the main body of the client is currently written in Scala - this should be refactored to Java then we
+ * can more easily just share the pool part of the client instead of creating a wrapper for the whole client.
+ *
+ * @author <a href="http://tfox.org">Tim Fox</a>
+ */
+public class ClientHelper {
+
+  private static final String DS_LOCAL_MAP_NAME_BASE =  "__vertx.MySQLPostgreSQL.datasources.";
+
+  public static AsyncSQLClient getOrCreate(Vertx vertx, JsonObject config, String dataSourceName, boolean mySQL) {
```

Comment:
`getOrCreate` and `lookupClient` seem to be the exact same methods, just one forwarding to another. Looks redundant to me.